utils/generate_types/index.js

```diff
@@ -0,0 +1,392 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//@ts-check
+const path = require('path');
+const Source = require('../doclint/Source');
+const {chromium} = require('../../index');
+const Documentation = require('../doclint/check_public_api/Documentation');
+const PROJECT_DIR = path.join(__dirname, '..', '..');
+const fs = require('fs');
+const {parseOverrides} = require('./parseOverrides');
+const objectDefinitions = [];
+const handledMethods = new Set();
+/** @type {Documentation} */
+let documentation;
```

Comment:
WDYT about slowly rewriting all our utils into typescript? Having types will definitely help.

Reply:
We have types through JSDoc. I'm always against typescript, so I am biased. With the utils, is harder because the utils will depend on the library to be built and the library depends on the utils to be built, so you end up needing to require the npm playwright as a devDependency of playwright or having separate bootstrap logic. I'd rather avoid it.